src/utils.ts

```diff
@@ -230,30 +230,33 @@ export async function retry<T>(call: () => Promise<T>, times: number, delayS: nu
  * @param unsignedTx The unsigned transaction that this function will estimate
  * @param senderAddress The address that the transaction will be submitted from
  * @param provider A valid ethers provider - will be used to reason the gas price
+ * @param gasMarkup Represents a percent increase on the total gas cost. For example, 0.2 will increase this resulting value by a factor of 1.2
  * @param gasPrice A manually provided gas price - if set, this function will not resolve the current gas price
  * @returns The total gas cost to submit this transaction - i.e. gasPrice * estimatedGasUnits
  */
 export async function estimateTotalGasRequiredByUnsignedTransaction(
   unsignedTx: PopulatedTransaction,
   senderAddress: string,
   provider: providers.Provider | L2Provider<providers.Provider>,
+  gasMarkup: number,
   gasPrice?: BigNumberish
 ): Promise<BigNumberish> {
+  const gasTotalMultiplier = 1.0 + gasMarkup;
```

Comment:
Should we set reasonable constraints on gasMarkup? i.e. not < -1.0 and not > say 4.0?